src/mono/wasm/runtime/invoke-cs.ts

```diff
@@ -352,12 +352,20 @@ type BindingClosure = {
     isDisposed: boolean,
 }
 
-export function invoke_method_and_handle_exception(method: MonoMethod, args: JSMarshalerArguments): void {
+export function invoke_method_and_handle_exception(method: MonoMethod, args: JSMarshalerArguments, skipPushOperation?: boolean): void {
     assert_bindings();
     const fail_root = mono_wasm_new_root<MonoString>();
     try {
+        if (MonoWasmThreads && !skipPushOperation) {
+            // TODO in future, the generated code of JSExport could do this
+            runtimeHelpers.javaScriptExports.push_operation();
+        }
         const fail = cwraps.mono_wasm_invoke_method_bound(method, args, fail_root.address);
-        if (fail) throw new Error("ERR24: Unexpected error: " + monoStringToString(fail_root));
+        if (fail) runtimeHelpers.abort("ERR24: Unexpected error: " + monoStringToString(fail_root));
```

Comment:
should put pop operation above this just in case someone traps the abort and execution continues.

Reply:
`fail` is not user exception, it would only happen if something went really wrong with runtime. In such case even calling pop could be impossible.